### PR TITLE
Solitaire

### PR DESCRIPTION
**Pull request by trisDeveloper**

### CODE REVIEW

File: examples/solitaire/src/SolitairePuzzle.tsx [hunk 1]
```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
...
+
       return {
         ...prev,
         stockCards: shuffledCards.slice(28),
         tableauCards: cards,
       };
-    });
+    }); 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
```
Comment (NoodleOfDeath) [inline examples/solitaire/src/SolitairePuzzle.tsx]: `state` needs to be added as a dependency here

```suggestion
  }, [state]);
```

File: examples/solitaire/src/SolitairePuzzle.tsx [hunk 2]
```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
...
+      });
+
       return {
         ...prev,
         stockCards: shuffledCards.slice(28),
         tableauCards: cards,
       };
-    });
+    }); 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (NoodleOfDeath) [inline examples/solitaire/src/SolitairePuzzle.tsx:132]: you remove this after you add `state` as a dependency

```suggestion
```

File: examples/solitaire/src/SolitairePuzzle.tsx [hunk 3]
```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
...
       try {
-        const { state, history } = (typeof data === 'string' ? JSON.parse(data) : data) as { state: HistoryState; history: HistoryState[]; };
-        if (history.length > 0) {
-          setState(state);
-          setGameHistory(history);
-        } else {
-          throw new Error('Error loading history');
-        }
+        const history = typeof data === 'string' ? JSON.parse(data) : data;
+        setGameHistory(history);
```
Comment (NoodleOfDeath) [inline examples/solitaire/src/SolitairePuzzle.tsx:143]: you also need to set the current `state` in addition to the game history

File: examples/solitaire/src/SolitairePuzzle.tsx [hunk 4]
```diff
@@ -175,32 +187,35 @@ export const Puzzle = ({
...
         moveCardsAfterWin({
-          setSuitCards: (suitCards) => setState({ ...newState, suitCards: suitCards as CardProps[][] }),
-          setTableauCards: (tableauCards) => setState({ ...newState, tableauCards: tableauCards as CardProps[][] }),
+          setState,
           ...dtState,
         });
-        onSuccess();
+        setTimeout(() => {
+          onSuccess();
+        }, 1000);
```
Comment (NoodleOfDeath) [inline examples/solitaire/src/SolitairePuzzle.tsx:217]: nice, for the animation im guessing?

File: examples/solitaire/src/SolitairePuzzle.tsx [hunk 5]
```diff
@@ -223,31 +313,14 @@ export const Puzzle = ({
...
-                  const resetWasteCards = prev.wasteCards.map(card => ({ ...card, isFaceUp: false }));
-                  return {
-                    ...prev,
-                    stockCards: resetWasteCards.reverse(),
-                    wasteCards: [],
-                  };
-                }
-              });
-            } } />
+            onDrawCard={ handleOnDrawCard } />
```
Comment (NoodleOfDeath) [inline examples/solitaire/src/SolitairePuzzle.tsx:323]: nice, this definitely makes for cleaner code

**Review by NoodleOfDeath — COMMENTED** [review]
dope. going to run it first and then approve

**Review by NoodleOfDeath — APPROVED** [review]
Works great! Just add the state as a dependency and set the state when the game loads history

File: examples/solitaire/src/SolitairePuzzle.tsx (hunk 2 above)
Comment (trisDeveloper) [reply examples/solitaire/src/SolitairePuzzle.tsx:132]: Even when I add the state, it still throws a warning about a missing dependency: "onProgress." But if I add onProgress, things get weird.

File: examples/solitaire/src/SolitairePuzzle.tsx (hunk 3 above)
Comment (trisDeveloper) [reply examples/solitaire/src/SolitairePuzzle.tsx:143]: it also makes things weird

**Pull request merged**